src/cow-react/modules/swap/pure/EthFlow/EthFlowStepper/steps/Step3.tsx

```diff
@@ -0,0 +1,88 @@
+import React from 'react'
+import { Icon, Flag, Check } from 'react-feather'
+import styled from 'styled-components/macro'
+import { ExplorerLinkStyled, EthFlowStepperProps, SmartOrderStatus } from '..'
+import { StatusIconState } from '../StatusIcon'
+import { Step } from '../Step'
+
+const RefundMessage = styled.span`
+  color: #0d5ed9;
+`
+
+const ExpiredMessage = styled.span``
+
+export function Step3({ nativeTokenSymbol, tokenLabel, order, refund, cancelation }: EthFlowStepperProps) {
+  const { state, isExpired, rejectedReason } = order
+  const { isRefunded, refundTx } = refund
+  const { isCanceled, cancelationTx } = cancelation
+
+  const isIndexing = state === SmartOrderStatus.CREATION_MINED
+  const isIndexed = state === SmartOrderStatus.INDEXED
+  const isCreating = state === SmartOrderStatus.CREATING
+  const isFilled = state === SmartOrderStatus.FILLED
+
+  const isRefunding = !!refundTx && !isRefunded
+  const isCanceling = !!cancelationTx && !isCanceled
+
+  const expiredBeforeCreate = isExpired && (isCreating || isIndexing)
+
+  let label: string, stepStatus: StatusIconState, icon: Icon
+  if (expiredBeforeCreate) {
```

Comment:
```
if (isIndexing) {
    return {
        label: 'Receive ' + tokenLabel,
        status: 'not-started',
        icon: 'Flag'
    }
}

if (isFilled) {
    return {
        label: 'Received ' + tokenLabel,
        status: 'success',
        icon: 'Flag'
    }
}

...
```

Reply:
Happy to do something in this lines, I even thought about it when I created this Step, however, the config has the label, status, icon, but also the `details`.

Let me see if I can explain my thinking process here.

The construction of the details, and the label/status/icon follow different logics. 

I could make 2 functions, one returns the first part, and the other returns the detail, but both functions would need to either:
- Receive the original parameters, and derive the flags from them (so both of them would do a similar thing)
- Derive the flags from the parameters, and then pass them to this getConfig function (it becomes too cumbersome, and also, when I need to add new conditions, I need to modify the params, which i see it as adding friction)

So considering this, i opted for just a short function that would do this 2 things (first gets the label/icon/state, and the create the details). The refactored part is the one I moved to a generic `Step` component, so this `Step3` can still be fairly small. 

Let me know if the above makes sense. One thing I can do, is to separate more the logic from the presentation also in Step3. I think is overkill given the size of this component, but I will give it a try anyways

Reply:
I just gave it a tried to move some of the logic to a config function, so the presentation is simpler, but it feels that is overkill (and makes it harder to make modifications, imo)

I was doing this function, so the JSX can be a bit dumber
<img width="686" alt="image" src="https://user-images.githubusercontent.com/2352112/202407229-3b708c30-f55f-4113-b604-64daf7595cbf.png">

Maybe if you feel strong about this, we can see ways to improve it together

Reply:
```
const [label, stepStatus, icon] = (() => {
    if (expiredBeforeCreate) {
      return ['Receive ' + tokenLabel, 'pending', Flag]
    }
    if (isIndexing) {
      return ['Receive ' + tokenLabel, 'not-started', Flag]
    }
    if (isFilled) {
      return ['Received ' + tokenLabel, 'success', Check]
    }
    if (isCanceled || isRefunded) {
      return [nativeTokenSymbol + ' Refunded', 'success', Check]
    }
    if (isIndexed) {
      return ['Receive ' + tokenLabel, 'pending', Flag]
    }

    return ['Receive ' + tokenLabel, 'not-started', Flag]
  })()
```